sklearn/metrics/scorer.py

```diff
@@ -58,17 +126,47 @@ def __repr__(self):
                    "" if self._sign > 0 else ", greater_is_better=False",
                    self._factory_args(), kwargs_string))
 
+    def __call__(self, estimator, X, y_true, sample_weight=None):
+        """Evaluate predicted target values for X relative to y_true.
+
+        Parameters
+        ----------
+        estimator : object
+            Trained estimator to use for scoring. Must have a predict_proba
+            method; the output of that is used to compute the score.
+
+        X : array-like or sparse matrix
+            Test data that will be fed to estimator.predict.
+
+        y_true : array-like
+            Gold standard target values for X.
+
+        sample_weight : array-like, optional (default=None)
+            Sample weights.
+
+        Returns
+        -------
+        score : float
+            Score function applied to prediction of estimator on X.
+        """
+        return self._score(partial(_cached_call, None), estimator, X, y_true,
+                           sample_weight=sample_weight)
+
     def _factory_args(self):
         """Return non-default make_scorer arguments for repr."""
         return ""
 
 
 class _PredictScorer(_BaseScorer):
-    def __call__(self, estimator, X, y_true, sample_weight=None):
+    def _score(self, method_caller, estimator, X, y_true, sample_weight=None):
         """Evaluate predicted target values for X relative to y_true.
 
         Parameters
         ----------
+        method_caller: callable
+            Call estimator's method with args and kwargs, potentially caching
```

Comment:
I don't find this description clear. Maybe "returns predictions given an estimator, method name, and other arguments"